src/main/java/com/listywave/list/application/service/CommentService.java

```diff
@@ -33,21 +35,29 @@ public class CommentService {
     private final ListRepository listRepository;
     private final UserRepository userRepository;
     private final ReplyRepository replyRepository;
+    private final MentionRepository mentionRepository;
     private final CommentRepository commentRepository;
     private final ApplicationEventPublisher applicationEventPublisher;
 
-    public CommentCreateResponse create(Long listId, String content, Long loginUserId) {
-        User user = userRepository.getById(loginUserId);
+    public CommentCreateResponse create(Long listId, Long writerId, String content, List<Long> mentionIds) {
+        User writer = userRepository.getById(writerId);
         ListEntity list = listRepository.getById(listId);
+        List<Mention> mentions = toMentions(mentionIds);
 
-        Comment comment = Comment.create(list, user, new CommentContent(content));
-        Comment saved = commentRepository.save(comment);
+        Comment comment = commentRepository.save(new Comment(list, writer, new CommentContent(content), mentions));
 
-        applicationEventPublisher.publishEvent(AlarmEvent.comment(list, saved));
-        return CommentCreateResponse.of(saved, user);
+        applicationEventPublisher.publishEvent(AlarmEvent.comment(list, comment));
+        return CommentCreateResponse.of(comment, writer);
     }
 
-    public CommentFindResponse getComments(Long listId, int size, Long cursorId) {
+    private List<Mention> toMentions(List<Long> mentionIds) {
+        return mentionIds.stream()
+                .map(userRepository::getById)
+                .map(Mention::new)
+                .toList();
+    }
```

Comment:
해당 구현하신 부분은 맨션 대상이 N명인 경우 N번 만큼 쿼리가 날라갑니다. 어플리케이션과 DB간 통신은 적을 수록 좋으니 아래와 같은 방식으로 바꾸길 제안드립니다!

아래와 같은 방법은 Spirng Data JPA에서 제공하는 findAllById를 통해 사전에 mentionIds 만큼 IN 절을 이용하여 하나의 쿼리로 다 가져온 후 stream을 돌려 new Mention 객체를 만드는 방식입니다!

```suggestion
    private List<Mention> toMentions(List<Long> mentionIds) {
          List<User> mentionedUsers = userRepository.findAllById(mentionedIds);  
          return mentionedUsers.stream()
                             .map(Mention::new)
                             .toList();
    }
```

Reply:
구현할 때는 멘션을 많이들 하지 않을 것이라 생각하고 했는데, 생각해보니 그건 아무도 모르는 일이군요.
감사합니다 !!